repose-aggregator/tests/functional-tests/src/integrationTest/groovy/features/services/datastore/RemoteDatastoreServiceTest.groovy

```diff
@@ -0,0 +1,172 @@
+import static org.linkedin.groovy.util.concurrent.GroovyConcurrentUtils.waitForCondition
+
+@Category(Slow.class)
+class RemoteDatastoreServiceTest extends Specification {
+    @Shared
+    def repose1
+
+    @Shared
+    def repose2
+
+    @Shared
+    def remote
+
+    @Shared
+    def deproxy
+
+    @Shared
+    def repose1Port
+
+    @Shared
+    def repose2Port
+
+    @Shared
+    def reposeOneEndpoint
+
+    @Shared
+    def reposeTwoEndpoint
+
+    @Shared
+    def remoteDatastoreEndpoint
+
+    def setupSpec() {
+        repose1Port = PortFinder.instance.getNextOpenPort()
+        repose2Port = PortFinder.instance.getNextOpenPort()
+        def remotePort = PortFinder.instance.getNextOpenPort()
+        def datastorePort = PortFinder.instance.getNextOpenPort()
+        def targetPort = PortFinder.instance.getNextOpenPort()
+
+        reposeOneEndpoint = "http://localhost:$repose1Port"
+        reposeTwoEndpoint = "http://localhost:$repose2Port"
+        remoteDatastoreEndpoint = "http://localhost:$datastorePort"
+
+        deproxy = new Deproxy()
+        deproxy.addEndpoint(targetPort)
+
+        def launcherAndLog1 = startRepose('repose1', repose1Port, targetPort, datastorePort, true)
+        repose1 = launcherAndLog1.ReposeValveLauncher
+
+        def launcherAndLog2 = startRepose('repose2', repose2Port, targetPort, datastorePort, true)
+        repose2 = launcherAndLog2.ReposeValveLauncher
+
+        def launcherAndLogR = startRepose('remote', remotePort, targetPort, datastorePort, false)
+        remote = launcherAndLogR.ReposeValveLauncher
+
+        waitUntilReadyToServiceRequests(launcherAndLog1.ReposeLogSearch)
+        waitUntilReadyToServiceRequests(launcherAndLog2.ReposeLogSearch)
+        waitUntilReadyToServiceRequests(launcherAndLogR.ReposeLogSearch)
+    }
+
+    def cleanupSpec() {
+        deproxy?.shutdown()
+        repose1?.stop()
+        repose2?.stop()
+        remote?.stop()
+    }
+
+    static def startRepose(String subName, int reposePort, int targetPort, int datastorePort, boolean client) {
+        def testProperties = new TestProperties(RemoteDatastoreServiceTest.canonicalName.replace('.', '/') + "/$subName")
+        testProperties.setReposePort(reposePort)
+        testProperties.setTargetPort(targetPort)
+        def reposeConfigProvider = new ReposeConfigurationProvider(testProperties)
+        def reposeValveLauncher = new ReposeValveLauncher(reposeConfigProvider, testProperties)
+        reposeValveLauncher.enableDebug()
+        def reposeLogSearch = new ReposeLogSearch(testProperties.getLogFile())
+        reposeLogSearch.cleanLog()
+        def params = testProperties.getDefaultTemplateParams()
+        def runDir = testProperties.runDirectory
+        params.put('repose.artifact.directory', "$runDir/artifacts")
+        params.put('datastorePort', datastorePort)
+        reposeValveLauncher.configurationProvider.applyConfigs("common", params)
```

Comment:
I had it in there and it was pointed out that since this is the only test in this snowflake, then it probably wasn't needed. However, since you brought it up and I originally had it in there, then I think it should be there. Added back in.